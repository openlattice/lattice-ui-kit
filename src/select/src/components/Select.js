import React from 'react';
import ReactSelect, { createFilter } from 'react-select';
import { mergeDeep } from 'immutable';

import SelectController from './SelectController';
import { selectStyles } from '../../../style/select';

export const defaultProps = {
  filterOption: createFilter({ ignoreAccents: false }),
  menuPlacement: 'auto',
  styles: selectStyles,
};

/* eslint-disable react/jsx-props-no-spreading */
const Select = (props) => (
  <SelectController
      {...props} // eslint-disable-line indent
      render={((selectProps) => <ReactSelect {...mergeDeep(defaultProps, selectProps)} />)} />
);
/* eslint-enable */

export default Select;
