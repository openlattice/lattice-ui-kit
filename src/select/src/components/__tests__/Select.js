import React from 'react';
import { mount, shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import 'jest-styled-components';
import { StyledSelect, StyledCreatable } from '../Select';

describe('StyledSelect', () => {

  it('render matches snapshot', () => {
    const tree = shallow(<StyledSelect />)
    expect(toJson(tree)).toMatchSnapshot();
  });

  it('attrs classNamePrefix is set to "lattice-select"', () => {
    const wrapper = mount(<StyledSelect />);
    expect(wrapper.instance().attrs.classNamePrefix).toEqual('lattice-select');
  });

});

describe('StyledCreatable', () => {

  it('render matches snapshot', () => {
    const tree = shallow(<StyledCreatable />)
    expect(toJson(tree)).toMatchSnapshot();
  });

  it('attrs classNamePrefix is set to "lattice-select"', () => {
    const wrapper = mount(<StyledCreatable />)
    expect(wrapper.instance().attrs.classNamePrefix).toEqual('lattice-select');
  });
  
});
