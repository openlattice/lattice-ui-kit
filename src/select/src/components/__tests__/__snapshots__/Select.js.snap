// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StyledCreatable render matches snapshot 1`] = `
.c0 .lattice-select__control {
  min-height: 44px;
  border-radius: 3px;
  background-color: #f9f9fd;
  border: solid 1px #dcdce7;
  margin-top: 10px;
  box-shadow: 0 0 0 0px;
}

.c0 .lattice-select__control:hover {
  background-color: #f0f0f7;
  border: solid 1px #dcdce7;
}

.c0 .lattice-select__control.lattice-select__control--is-focused {
  border: solid 1px #6124e2;
  box-shadow: 0 0 0 0px;
  background-color: #fff;
}

.c0 .lattice-select__option {
  color: #555e6f;
  font-size: 14px;
  line-height: 19px;
}

.c0 .lattice-select__option:active {
  background-color: #e4d8ff;
}

.c0 .lattice-select__option--is-focused {
  background-color: #f0f0f7;
}

.c0 .lattice-select__option--is-selected {
  background-color: #e6e6f7;
  color: #6124e2;
}

.c0 .lattice-select__single-value {
  color: #2e2e34;
  font-size: 14px;
  line-height: 19px;
}

.c0 .lattice-select__indicator-separator {
  display: none;
}

<StateManager
  className="c0"
  classNamePrefix="lattice-select"
  defaultInputValue=""
  defaultMenuIsOpen={false}
  defaultValue={null}
/>
`;

exports[`StyledSelect render matches snapshot 1`] = `
.c0 .lattice-select__control {
  min-height: 44px;
  border-radius: 3px;
  background-color: #f9f9fd;
  border: solid 1px #dcdce7;
  margin-top: 10px;
  box-shadow: 0 0 0 0px;
}

.c0 .lattice-select__control:hover {
  background-color: #f0f0f7;
  border: solid 1px #dcdce7;
}

.c0 .lattice-select__control.lattice-select__control--is-focused {
  border: solid 1px #6124e2;
  box-shadow: 0 0 0 0px;
  background-color: #fff;
}

.c0 .lattice-select__option {
  color: #555e6f;
  font-size: 14px;
  line-height: 19px;
}

.c0 .lattice-select__option:active {
  background-color: #e4d8ff;
}

.c0 .lattice-select__option--is-focused {
  background-color: #f0f0f7;
}

.c0 .lattice-select__option--is-selected {
  background-color: #e6e6f7;
  color: #6124e2;
}

.c0 .lattice-select__single-value {
  color: #2e2e34;
  font-size: 14px;
  line-height: 19px;
}

.c0 .lattice-select__indicator-separator {
  display: none;
}

<StateManager
  className="c0"
  classNamePrefix="lattice-select"
  defaultInputValue=""
  defaultMenuIsOpen={false}
  defaultValue={null}
/>
`;
